### PR TITLE
close #2014 - finished TransitionSpec

### LINKED ISSUE #2014: Port Akka.Cluster MultiNodeSpec: TransitionSpec
https://github.com/akka/akka/blob/master/akka-cluster/src/multi-jvm/scala/akka/cluster/TransitionSpec.scala

### PR DESCRIPTION
**Pull request by Aaronontheweb**

### CODE REVIEW

File: src/core/Akka.Cluster.Tests.MultiNode/TransitionSpec.cs [hunk 1]
```diff
@@ -87,32 +87,38 @@ private ImmutableHashSet<Address> MemberAddresses()
...
         {
-            // TODO: is it proper conversion of clusterView.seenBy flatMap roleName?
-            return ClusterView.SeenBy.Select(c => RoleName(c)).ToImmutableHashSet();
+            return ClusterView.SeenBy.Select(RoleName).ToImmutableHashSet();
         }
 
         private void AwaitSeen(params Address[] addresses)
         {
-            SeenLatestGossip().Select(c => GetAddress(c)).Should().BeEquivalentTo(addresses.ToImmutableHashSet());
+            AwaitAssert(() =>
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster.Tests.MultiNode/TransitionSpec.cs]: All of these methods needed to have `AwaitAssert` used inside of them in order to pass.

File: src/core/Akka.Cluster.Tests.MultiNode/TransitionSpec.cs [hunk 2]
```diff
@@ -153,16 +159,16 @@ private void GossipTo(RoleName fromRole, RoleName toRole)
             {
                 EnterBarrier("before-gossip-" + _gossipBarrierCounter);
                 EnterBarrier("after-gossip-" + _gossipBarrierCounter);
-            }, Roles.Where(r => r != fromRole || r != toRole).ToArray());
+            }, Roles.Where(r => r != fromRole && r != toRole).ToArray());
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster.Tests.MultiNode/TransitionSpec.cs]: Bug that caused the toRole to accidentally enter the same barriers twice

File: src/core/Akka.Cluster/ClusterDaemon.cs [hunk 3]
```diff
@@ -1907,7 +1907,7 @@ public void AssertLatestGossip()
         public void Publish(Gossip newGossip)
         {
             _publisher.Tell(new InternalClusterAction.PublishChanges(newGossip));
-            if (_cluster.Settings.PublishStatsInterval == TimeSpan.MinValue)
+            if (_cluster.Settings.PublishStatsInterval == TimeSpan.Zero)
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/ClusterDaemon.cs]: Defaults to `Timespan.Zero`, not `Timespan.MinValue`.

File: src/core/Akka.Cluster/ClusterDaemon.cs [hunk 4]
```diff
@@ -2143,6 +2143,7 @@ public GossipStats IncrementOlderCount()
         {
             return new GossipStats(receivedGossipCount ?? ReceivedGossipCount,
                 mergeCount ?? MergeCount,
+                sameCount ?? SameCount,
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/ClusterDaemon.cs]: weren't copying `sameCount` before

File: src/core/Akka.Cluster/ClusterDaemon.cs (hunk 3 above)
Comment (alexvaluyskiy) [inline src/core/Akka.Cluster/ClusterDaemon.cs]: Good catch

**Pull request merged**